client/keys/add.go

```diff
@@ -101,16 +103,23 @@ func runAddCmd(cmd *cobra.Command, args []string) error {
 		kb = keys.NewInMemory()
 		encryptPassword = DefaultKeyPass
 	} else {
-		kb, err = NewKeyBaseFromHomeFlag()
-		if err != nil {
-			return err
+		// c.Flags().Bool(FlagSecretSore, false, "Use legacy secret store")
+		if viper.GetBool(flags.FlagSecretStore) {
+			os.Stderr.WriteString("Using deprecated secret store. This will be removed in a future release.")
```

Comment:
Rather than stating it'll be removed in a future release, we should state: `use at your own risk as support is likely to be removed without warning`.